app/models/activity_submission.rb

```diff
@@ -20,4 +23,17 @@ def github_url_required?
     activity && activity.allow_submissions?
   end
 
+  def request_code_review
+    if self.activity.allow_submissions? && should_code_review?
+      CodeReviewRequest.create(activity_submission: self, requestor_id: self.user.id)
+    end
+  end
+
+  def should_code_review?
+    # TODO Vary these probabilities
+    student_probablitiy = 0.8
```

Comment:
I would rather that these be in the model as integer (0 to 100) that `:default` (in migration) to 80 and 60 accordingly.